api/tests/unit/environments/identities/test_unit_identities_feature_states_views.py

```diff
@@ -106,3 +112,121 @@ def test_user_with_view_environment_permission_can_retrieve_all_feature_states_f
+        string_value=source_feature_state_1_value
+    )
+
+    source_feature_state_2: FeatureState = FeatureState.objects.create(
+        feature=feature_2,
+        environment=environment,
+        identity=source_identity,
+        enabled=True,
+    )
+    source_feature_state_2_value = "Source Identity for feature value 2"
+    FeatureStateValue.objects.filter(feature_state=source_feature_state_2).update(
+        string_value=source_feature_state_2_value
+    )
+
+    # Create 2 feature states for target identity for feature 2 and feature 3
+    target_feature_state_2: FeatureState = FeatureState.objects.create(
+        feature=feature_2,
+        environment=environment,
+        identity=target_identity,
+        enabled=False,
+    )
+    target_feature_state_2_value = "Target Identity value for feature 2"
+    FeatureStateValue.objects.filter(feature_state=target_feature_state_2).update(
+        string_value=target_feature_state_2_value
+    )
+
+    FeatureState.objects.create(
+        feature=feature_3,
+        environment=environment,
+        identity=target_identity,
+        enabled=False,
+    )
+
+    clone_identity_feature_states_url = reverse(
+        "api-v1:environments:identity-featurestates-clone-from-given-identity",
+        args=[environment.api_key, target_identity.id],
+    )
+
+    get_feature_states_url = reverse(
+        "api-v1:environments:identity-featurestates-all",
+        args=[environment.api_key, target_identity.id],
+    )
+
+    # WHEN
+
+    clone_identity_feature_states_response = admin_client.post(
+        clone_identity_feature_states_url,
+        data=json.dumps({"source_identity_id": str(source_identity.id)}),
+        content_type="application/json",
+    )
+
+    # And getting feature states for target identity
+    get_feature_states_response = admin_client.get(get_feature_states_url)
+
+    # THEN
+
+    # Assert API calls responses status are HTTP OK (200)
+    assert clone_identity_feature_states_response.status_code == status.HTTP_200_OK
+    assert get_feature_states_response.status_code == status.HTTP_200_OK
+
+    response = get_feature_states_response.json()
+
+    # Assert target identity contains only the 2 cloned overridden features states and 1 environment feature state
+    assert len(response) == 3
+
+    # Assert cloned data is correct
+    assert response[0]["feature"]["id"] == feature_1.id
+    assert response[0]["enabled"] == source_feature_state_1.enabled
+    assert response[0]["feature_state_value"] == source_feature_state_1_value
+    assert response[0]["overridden_by"] == "IDENTITY"
+
+    assert response[1]["feature"]["id"] == feature_2.id
+    assert response[1]["enabled"] == source_feature_state_2.enabled
+    assert response[1]["feature_state_value"] == source_feature_state_2_value
+    assert response[1]["overridden_by"] == "IDENTITY"
+
+    assert response[2]["feature"]["id"] == feature_3.id
+    assert response[2]["overridden_by"] is None
```

Comment:
Again I'd add tests to this response record for `enabled` and `feature_state_value`

Reply:
That is not part of this changes and below we verify that the feature state has been removed.